File: Python/ anagram.py
class Solution:
    # @param strs, a list of strings
    # @return a list of strings
    def anagrams(self, strs):
       d = collections.defaultdict(list)
       for s in strs:
           d[tuple(sorted(s))].append(s)  
       return [a for agram_group in d.values() if len(agram_group)>1 for a in agram_group]

File: Java/src/ anagram.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.LinkedList;

public class Solution
{
  public List<String> anagrams(String[] strs)
  {
    Map<String, List<String>> dict = new HashMap<>();
    List<String> ret = new LinkedList<>();

    for (String s : strs)
    {
      char[] key = s.toCharArray();
      Arrays.sort(key);
      String dictKey = new String(key);
      if (dict.containsKey(dictKey))
      {
        dict.get(dictKey).add(s);
      }
      else
      {
        List<String> strList = new ArrayList<>();
        strList.add(s);
        dict.put(dictKey, strList);
      }
    }

    for (Map.Entry<String, List<String>> entry : dict.entrySet())
    {
      List<String> value = entry.getValue();

      if (value.size() > 1)
      {
        ret.addAll(value);
      }
    }
    return ret;
  }
}
